syntax = "proto3";

package myapp;

service DatabaseService {
  // Create operation
  rpc CreateRecord (CreateRequest) returns (CreateResponse);

  // Read operation
  rpc ReadRecord (ReadRequest) returns (ReadResponse);

  // Update operation
  rpc UpdateRecord (UpdateRequest) returns (UpdateResponse);

  // Delete operation
  rpc DeleteRecord (DeleteRequest) returns (DeleteResponse);
}

message CreateRequest {
  string name = 1;
  string description = 2;
  string status = 3;
}

message CreateResponse {
  string message = 1;
}

message ReadRequest {
  string record_id = 1;
}

message ReadResponse {
  string data = 1;
}

message UpdateRequest {
  string record_id = 1;
  string name = 2;
  string description = 3;
  string status = 4;
}

message UpdateResponse {
  string message = 1;
}

message DeleteRequest {
  string record_id = 1;
}

message DeleteResponse {
  string message = 1;
}
